cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -35,7 +35,81 @@ endmacro()
 # Build, link and optionally package an add-on
 macro (build_addon target prefix libs)
   addon_version(${target} ${prefix})
+
   if(${prefix}_SOURCES)
+    # Read used headers from addon, needed to identitfy used kodi addon interface headers
+    if(${prefix}_HEADERS)
+      # Add the used header files defined with CMakeLists.txt from addon itself
+      foreach(hdr_file ${${prefix}_HEADERS})
+        list(APPEND USED_SOURCES ${PROJECT_SOURCE_DIR}/${hdr_file})
+      endforeach()
+    else()
+      message(AUTHOR_WARNING "You have not defined used headers in cmake, scanning all addon included files!")
```

Comment:
"Header files not defined in your CMakeLists.txt. Please consider defining ${prefix}_HEADERS as list of all headers used by this addon. Falling back to recursive scan for *.h"